gradle-conjure/build.gradle

```diff
@@ -24,7 +24,8 @@ dependencies {
     compile project(':gradle-conjure-api')
     compile 'com.google.guava:guava'
     compile 'com.palantir.sls.versions:sls-versions'
-    compile 'com.fasterxml.jackson.core:jackson-databind'
+    compile 'com.palantir.conjure.java.runtime:conjure-java-jackson-serialization'
+    compile 'com.palantir.sls-packaging:gradle-recommended-product-dependencies'
```

Comment:
might as well use `implementation` for these? I know nobody should be compiling against these, but worth sticking to best practises?